_posts/2021-06-23-cockpit-247.md

```diff
@@ -0,0 +1,56 @@
+---
+title: Cockpit 247
+author: kkoukiou
+date: 2021-06-23
+tags: cockpit linux
+slug: cockpit-247
+category: release
+summary: Cockpit with Metrics and Virtual Machines improvements
+---
+
+Cockpit is the [modern Linux admin interface](https://cockpit-project.org/).  We release regularly.
+
+Here are the release notes from Cockpit version 247 and Cockpit Machines version 246.
+
+## Metrics: Enable Grafana client setup
+
+The "Metrics settings" button introduced in [Cockpit 245](https://cockpit-project.org/blog/cockpit-245.html)
+has a new switch to set up exporting metrics to the network through the [PCP pmproxy service](https://linux.die.net/man/1/pmproxy)
+and the [Redis database](https://grafana-pcp.readthedocs.io/en/latest/datasources/redis.html):
+
+![metrics settings](/images/metrics-settings-2.png)
+
+If the machine uses firewalld, you can allow pmproxy in a particular zone afterwards:
+
+![metrics firewall zone selector](/images/metrics-fw-sel.png)
+
+After that, a remote [Grafana](https://grafana.com/) instance can read the machine's metrics through the
+[PCP Grafana plugin](https://grafana-pcp.readthedocs.io/), as described on the [PCP metrics feature page](https://cockpit-project.org/guide/latest/feature-pcp.html).
+
+## Machines: Share host files with the guest using virtiofs
+
+The user can now share a host directory tree with a guest. This is internally using [virtiofs](https://virtiofs.gitlab.io/),
+which is designed to offer local file system semantics and performance.
+virtiofs is available in mainline since Linux 5.4, QEMU 5.0, libvirt 6.2 and Web console will not expose this feature in the UI if the requirements are not met.
```

Comment:
```suggestion
Cockpit Machines can now share a host directory tree with a guest VM. 

[virtio-fs](https://virtiofs.gitlab.io/) powers this sharing feature, enabling local file system semantics with performance. virtio-fs has been available in mainline Linux 5.4 (November 2019), QEMU 5.0 (April 2020), and libvirt 6.2 (April 2020). Sharing host files is available only when these versions are met.
```

Reply:
'enabling local file system semantics with performance'  <- this is different meaning than the original sentence.

Reply:
also I would avoid including the dates because different distros instroduce these package version at different points in time.

Reply:
> also I would avoid including the dates because different distros instroduce these package version at different points in time

Hmm, right. It's the release dates of each to give context that it isn't a new thing.

```suggestion
Cockpit Machines can now share a host directory tree with a guest VM. 

[virtio-fs](https://virtiofs.gitlab.io/) powers this sharing feature, with local file system semantics and performance. virtio-fs has been available in mainline Linux 5.4, QEMU 5.0, and libvirt 6.2. Sharing host files is available only when these versions are met.
```

Reply:
we agreed to use virtiofs - after the developers contacted me second time about the wording.

Reply:
Also instead of 'with local... ' can we still use 'offering local...'

Since it's not clear the the underlying tool is implemented in that way, and not cockpit itself.

Reply:
```suggestion
Cockpit Machines can now share a host directory tree with a guest VM.

[virtiofs](https://virtiofs.gitlab.io/) powers this sharing feature, offering local file system semantics and performance. virtiofs has been available in mainline Linux 5.4, QEMU 5.0, and libvirt 6.2. Sharing host files is available only when these versions are met.
```